cleaning up example workflow for GW loss

janetrbarclay opened the pull request:
This cleans up and fixes some errors in the example workflow for the GW loss. In particular, it removes the dependencies on the StatsModels and Seaborn packages. 

The test data for this workflow is the data for the Christiana. I got errors with the rgcn test data, I suspect b/c that doesn't included segments / times for which we can calculate the gw loss, though I didn't actually test that.

SimonTopp commented on river_dl/gw_utils.py:
```diff
@@ -1,12 +1,13 @@
 import numpy as np
 import xarray as xr
-import statsmodels.api as sm
+#import statsmodels.api as sm
```
Maybe just delete?
```suggestion
```

SimonTopp commented on river_dl/gw_utils.py:
```diff
@@ -1,12 +1,13 @@
 from itertools import compress, product
 import matplotlib.pyplot as plt
-import seaborn as sns
+#import seaborn as sns
```
Same here, unless you think you'll still use it somewhere.
```suggestion
```

SimonTopp commented on river_dl/gw_utils.py:
```diff
@@ -42,38 +43,53 @@ def amp_phi (Date, temp, isWater=False, r_thresh=0.8, tempType="obs"):
     try:
-        model = sm.OLS(temp,X, missing='drop')
-        results = model.fit()
+        #this solves the regression using stats models, which provides confidence intervals on the coefficients
+        #X = sm.add_constant(x) #use this with statsmodel
+        #model = sm.OLS(temp,X, missing='drop')
+        #results = model.fit()
```
Same as above, if you think there's a reason to keep all the `statsmodel` code in here we can, but unless we think we'll be jumping back and forth I'd just remove it.

SimonTopp commented on river_dl/gw_utils.py:
```diff
@@ -42,38 +43,53 @@ def amp_phi (Date, temp, isWater=False, r_thresh=0.8, tempType="obs"):
+        #model = sm.OLS(temp,X, missing='drop')
+        #results = model.fit()
+        
+        # this solves the regression using scikit-learn (not on the current import list), which doesn't give confidence intervals
```
```suggestion
        # this solves the regression using scikit-learn, which doesn't give confidence intervals
```

SimonTopp reviewed (approved):
This all looks good to me! My one comment would be to delete all the legacy code rather than commenting it out, but if you think you might end up jumping back forth between them then I can understand why you'd leave it in.  Since you use this the most I'm happy to defer to you on that.

janetrbarclay commented:
Thanks! I removed the old seaborn code. I left the StatsModel bits. I had switched to StatsModel because it gives us the ability to calculate confidence intervals for the Ar and deltaPhi (which sklearn doesn't). We're not using this now, but I'm leaving the old code in case we want to reinstate that function.
